app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModel.kt

```diff
@@ -22,8 +20,7 @@ class PartiesViewModel @Inject constructor(
 	private val partiesRepository: PartiesRepository,
 	@Validator.ShortText private val shortTextValidator: TextValidator,
 	@Validator.LongText private val longTextValidator: TextValidator,
-) : MVIViewModel<PartiesState, PartiesEvent>(initialState = PartiesState()),
-	SideEffectEmitter<PartiesSideEffect> by SideEffectEmitterImplementation() {
+) : MVIViewModel<PartiesState, PartiesEvent, PartiesSideEffect>(initialState = PartiesState()) {
 
 	init {
 		loadParties()
```

Comment:
This should be handled by event to follow the MVI approach

Reply:
How do you see it? Emit some event from view at the beginning to initialize the VM? Could you elaborate more because for me loading something in the constructor is much simpler than sending event to vw to load initial data on screen load.

Reply:
Agree with @BorysChajdas , I got you @MarekMacko it will be concise with MVI, but I think the patterns are for developers and we can be agile and not stick to the rules always

Reply:
@MarekMacko  do You mean here, to emit Load event on init instead of loadParties() usage ?

Reply:
@marcinWisnia it would be the easiest, but it would break the pattern as the events should be emitted by view

Reply:
So some state is by constructor and some by event? Note that creation, and calling the constructor, is determined by the view. In the same way we can handle some clicks by `viewModel::onClick` and some by sending events. 
How would you like to handle call rest of the lifecycle events then (onStop, back clicks, etc.) ?
I think that everyone can predict what would happen with the code after few months :)